devices/sierra_wireless/wp7702/impl.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2019-2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include "impl.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#define MAXLINE 1024
+#define IMSI_LEN 15
+#define AES_BLOCK_SIZE 16
+
+static retcode_t wp7702_get_key(uint8_t *key);
+static retcode_t wp7702_get_hardware_identifier(char *device_id);
+static void wp7702_release(device_t *);
+
+device_t *new_wp7702(void) {
```

Comment:
Since we are going to implement abstract interface, we should avoid exposing functions such as `new_wp7702`.

Reply:
Thinking of the case in Linux VFS, Linux has many file systems but there are only descriptors exposed.